### PR TITLE
Parse After Effects blur effects from Lottie files.

### PR DESCRIPTION
**Pull request by simeoncran**
Blur is not yet supported in the translator.

### CODE REVIEW

File: source/LottieData/Enum.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            _value = value;
+        }
+
+        public T Value => _value;
+
+        public static implicit operator Enum<T>(T value) => new Enum<T>(value);
+
+        public static implicit operator T(Enum<T> value) => value;
+
+        public bool Equals(Enum<T> other) => other._value.CompareTo(_value) == 0;
```
Comment (derobinsMSFT) [inline source/LottieData/Enum.cs:36]: >_value [](start = 51, length = 6)

`Value`?

File: source/LottieData/GaussianBlurEffect.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.Toolkit.Uwp.UI.Lottie.LottieData
+{
+    /// <summary>
+    /// A gaussian blur effect.
```
Comment (derobinsMSFT) [inline source/LottieData/GaussianBlurEffect.cs:8]: >g [](start = 10, length = 1)

Nit: capitalize.

File: source/LottieReader/Serialization/Animatables.cs [hunk 3]
```diff
@@ -28,6 +30,15 @@ sealed partial class LottieCompositionReader
         static readonly AnimatableVector3 s_animatableVector3Zero = new AnimatableVector3(Vector3.Zero, null);
         static readonly AnimatableVector3 s_animatableVector3OneHundred = new AnimatableVector3(new Vector3(100, 100, 100), null);
 
+        static readonly AnimatableParser<Enum<BlurDimension>> s_animatableBlurDimensionParser =
+            CreateAnimatableParser((in LottieJsonElement element) => element.AsInt32() switch
+            {
+                1 => (Enum<BlurDimension>)BlurDimension.HorizontalAndVertical,
```
Comment (derobinsMSFT) [inline source/LottieReader/Serialization/Animatables.cs:36]: >BlurDimension [](start = 42, length = 13)

Can the ordering of the `BlurDimension` values be changed to line up with the `element` values they're derived from?

File: source/LottieReader/Serialization/Effects.cs [hunk 4]
```diff
@@ -60,6 +62,73 @@ Effect[] ReadEffectsList(in LottieJsonArrayElement? array, string layerName)
...
+            }
+
+            return new GaussianBlurEffect(
+                effectName,
+                isEnabled,
+                blurriness: blurriness,
+                blurDimensions: blurDimensions,
+                repeatEdgePixels: repeatEdgePixels);
+
+            static Exception ParseFailure() => ReaderException("Invalid gaussian blur effect");
```
Comment (derobinsMSFT) [inline source/LottieReader/Serialization/Effects.cs:128]: >effect [](start = 86, length = 6)

Nit: period.

Comment (derobinsMSFT) [inline source/LottieReader/Serialization/Effects.cs:128]: >g [](start = 72, length = 1)

Nit: capitalize.

File: source/LottieData/Enum.cs (hunk 1 above)
Comment (simeoncran) [reply source/LottieData/Enum.cs:36]: Could be, but why?

---
In reply to: [529110015](https://github.com/windows-toolkit/Lottie-Windows/pull/393#discussion_r529110015) [](ancestors = 529110015)

File: source/LottieReader/Serialization/Animatables.cs (hunk 3 above)
Comment (simeoncran) [reply source/LottieReader/Serialization/Animatables.cs:36]: They could however they are independent. The encoding of the values is encapsulated in parsers and serializers and shouldn't leak into the enum values. Chat if you disagree - to save me preemptively typing out a defense for this position. :-)

---
In reply to: [529111458](https://github.com/windows-toolkit/Lottie-Windows/pull/393#discussion_r529111458) [](ancestors = 529111458)

**Review by derobinsMSFT — APPROVED** [review]
:shipit:

**Pull request merged**